src/service.rs

```diff
@@ -831,16 +825,18 @@ impl Service {
 
     /// Sends a PING request to a node.
     fn send_ping(&mut self, enr: Enr) {
-        let request_body = RequestBody::Ping {
-            enr_seq: self.local_enr.read().seq(),
-        };
-        let active_request = ActiveRequest {
-            contact: enr.into(),
-            request_body,
-            query_id: None,
-            callback: None,
-        };
-        self.send_rpc_request(active_request);
+        if let Ok(contact) = NodeContact::try_from_enr(enr) {
+            let request_body = RequestBody::Ping {
+                enr_seq: self.local_enr.read().seq(),
+            };
+            let active_request = ActiveRequest {
+                contact,
+                request_body,
+                query_id: None,
+                callback: None,
+            };
+            self.send_rpc_request(active_request);
+        }
```

Comment:
Maybe we log a warn here, so we can see if there are non-contactable nodes in our routing table. It shouldn't be the case, but being noisy will help identify them